node/pkg/node/options.go

```diff
@@ -55,33 +55,38 @@ func GuardianOptionP2P(p2pKey libp2p_crypto.PrivKey, networkId, bootstrapPeers,
 			// Add the gossip advertisement address
 			components.GossipAdvertiseAddress = gossipAdvertiseAddress
 
-			g.runnables["p2p"] = p2p.Run(
-				g.obsvC,
-				g.obsvReqC.writeC,
-				g.obsvReqSendC.readC,
-				g.gossipSendC,
-				g.signedInC.writeC,
+			params, err := p2p.NewRunParams(
+				bootstrapPeers,
+				networkId,
 				p2pKey,
-				g.gk,
 				g.gst,
-				networkId,
-				bootstrapPeers,
-				nodeName,
-				disableHeartbeatVerify,
```

Comment:
I think this one got lost. Not sure if that was intentional.

Reply:
Good catch. I added it to `WithGuardianOptions`.